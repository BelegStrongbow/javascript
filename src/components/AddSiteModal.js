import React from "react";
import Modal from "react-modal";
import styled from "styled-components";
import { defineMessages, injectIntl, intlShape } from "react-intl";
import AddSite from "../components/AddSite";

const messages = defineMessages( {
	modalAriaLabel: {
		id: "modal.arialabel.add",
		defaultMessage: "Add a new site",
	},
} );

/*
 * Makes the `aria-hidden="true"` attribute being applied on the root element
 * instead of the body.
 */
Modal.setAppElement( "#root" );

class BaseAddSiteModal extends React.Component {

	constructor( props ) {
		super( props );
	}

	/**
	 * Returns the rendered html.
	 *
	 * @returns {ReactElement} The rendered html.
	 */
	render() {
		console.log( this.props.onClose );
		return (
			<div>
				<Modal
					isOpen={ this.props.isOpen }
					onRequestClose={ this.props.onClose }
					role="dialog"
					contentLabel={ this.props.intl.formatMessage( messages.modalAriaLabel ) }
					overlayClassName={ `${ this.props.className } my-yoast-modal__overlay` }
					className={ `${ this.props.className } my-yoast-modal__content` }
				>
					<AddSite
						onLinkClick={ this.props.onLink }
						onCancelClick={ this.props.onClose }
						onChange={ this.props.onChange }
						errorFound={ this.props.errorFound }
						errorMessage= { this.props.errorMessage }
					/>
				</Modal>
			</div>
		);
	}
}

BaseAddSiteModal.propTypes = {
	className: React.PropTypes.string,
	intl: intlShape.isRequired,
	isOpen: React.PropTypes.bool,
	onClose: React.PropTypes.func.isRequired,
	onLink: React.PropTypes.func.isRequired,
	onChange: React.PropTypes.func.isRequired,
	errorFound: React.PropTypes.bool.isRequired,
	errorMessage: React.PropTypes.string,
};

BaseAddSiteModal.defaultProps = {
	isOpen: false,
	errorMessage: "",
};

const AddSiteModal = styled( BaseAddSiteModal )`
	&.my-yoast-modal__overlay {
		position: fixed;
		top: 0;
		left: 0;
		right: 0;
		bottom: 0;
		background-color: rgba(0, 0, 0, 0.6);
	}

	&.my-yoast-modal__content {
		position: absolute;
		top: 50%;
		left: 50%;
		right: auto;
		bottom: auto;
		width: auto;
		border: 0;
		border-radius: 0;
		margin-right: -50%;
		padding: 40px 20px;
		transform: translate(-50%, -50%);
		background-color: #fff;
		outline: none;
		
		@media screen and ( max-width: 500px ) {
			height: 300px;
			width: auto;
			overflow: scroll;
		}
		
		@media screen and ( max-height: 640px ) {
			height: 300px;
			width: auto;
			overflow: scroll;
		}

	}

	.my-yoast-modal__actions {
		padding-top: 1em;
		text-align: right;
	}

	.my-yoast-modal__actions button {
		margin-left: 1em;
	}
`;

export default injectIntl( AddSiteModal );
