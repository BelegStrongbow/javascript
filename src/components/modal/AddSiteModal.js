import { defineMessages } from "react-intl";
import MyYoastModal from "../MyYoastModal";
import AddSite from "../AddSite";
import React from "react";
import PropTypes from "prop-types";

let modalAriaLabel = defineMessages( {
	id: "modal.arialabel.add",
	defaultMessage: "Add a new site",
} );

/**
 * Returns the AddSite modal.
 *
 * @param {Object} props The props required.
 * @returns { ReactElement } The AddSite Modal.
 */
const AddSiteModal = ( props ) => (
	<MyYoastModal
		isOpen={ props.modalOpen }
		onClose={ props.onClose }
		modalAriaLabel={ modalAriaLabel }
	>
		<AddSite
			onConnectClick={ props.onConnect }
			onCancelClick={ props.onClose }
			onChange={ props.onChange }
			errorFound={ props.errorFound }
			error={ props.error }
			query={ props.query }
			linkingSiteUrl={ props.linkingSiteUrl }
		/>
	</MyYoastModal>
);


AddSiteModal.propTypes = {
	onConnect: PropTypes.func.isRequired,
	onClose: PropTypes.func.isRequired,
	onChange: PropTypes.func.isRequired,
	errorFound: PropTypes.bool.isRequired,
	error: PropTypes.object,
	linkingSiteUrl: PropTypes.string.isRequired,
	query: PropTypes.string,
	modalOpen: PropTypes.bool,
};

AddSiteModal.defaultProps = {
	query: "",
	linkingSiteUrl: "",
	modalOpen: false,
	error: null,
};

export default AddSiteModal;
