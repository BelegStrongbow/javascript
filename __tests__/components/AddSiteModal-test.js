import React from 'react';
import { createComponentWithIntl } from "../../utils";

import AddSiteModal from '../../src/components/AddSiteModal';

jest.mock( "react-modal", () => {
	function Modal(props) {
		return <div id="Modal" {...props}>{props.children}</div>
	}

	Modal.setAppElement = jest.fn();

	return Modal;
} );

test('The AddSiteModal component matches the snapshot', () => {
	const component = createComponentWithIntl(
		<AddSiteModal onClose={ () => {
			console.log( "clicked on Cancel" );
		} } onLink={ () => {
			console.log( "clicked on Link" );
		} } isOpen={ true }
			onChange={ () => {} }
			errorFound={ false }
			query=""
		/>

	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('the addSiteModal handling an onClose event', () => {
	const component = createComponentWithIntl(
		<AddSiteModal onClose={ () => {
			console.log( "clicked on Cancel" );
		} } onLink={ () => {
			console.log( "clicked on Link" );
		} } isOpen={ true }
			onChange={ () => {} }
			errorFound={ false }
			query=""
		/>
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();

	// manually trigger the callback.
	tree.children[0].props.onRequestClose();

	// re-rendering
	tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test('the addSiteModal handling an onLink event', () => {
	const component = createComponentWithIntl(
		<AddSiteModal onClose={ () => {
			console.log( "clicked on Cancel" );
		} } onLink={ () => {
			console.log( "clicked on Link" );
		} } isOpen={ true }
			onChange={ () => {} }
			errorFound={ false }
			query=""
		/>
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();

	// manually trigger the callback.
	tree.children[0].children[0].children[3].children[1].props.onClick();

	// re-rendering
	tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
