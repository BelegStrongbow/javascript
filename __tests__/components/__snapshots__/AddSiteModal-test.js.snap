exports[`test The AddSiteModal component matches the snapshot 1`] = `
<div>
  <div
    className="hBejNq my-yoast-modal__content"
    contentLabel="Add a new site"
    id="Modal"
    isOpen={true}
    onRequestClose={[Function]}
    overlayClassName="hBejNq my-yoast-modal__overlay"
    role="dialog">
    <div
      className="hfMDPK">
      <h1
        className="dQgKWp">
        <span>
          Add Site
        </span>
      </h1>
      <p
        className="hUuGBg">
        <label
          htmlFor="addSiteInputField">
          <span>
            Please enter the URL of the site you would like to link with your account:
          </span>
        </label>
      </p>
      <input
        aria-describedby="addSiteInfo"
        className="jssTrf"
        defaultValue=""
        id="addSiteInputField"
        onChange={[Function]}
        placeholder="https://example-site.com"
        type="url" />
      <div
        className="gCxKqx">
        <div
          className="iMrCDo">
          <button
            className="cAeNWO"
            onClick={[Function]}
            type="button">
            <span>
              cancel
            </span>
          </button>
          <button
            className="gxQXtW"
            onClick={[Function]}
            type="button">
            <span>
              connect
            </span>
          </button>
        </div>
      </div>
      <div
        className="bPgglV" />
      <img
        alt=""
        className="ireVCO"
        src="addsite.svg" />
    </div>
  </div>
</div>
`;

exports[`test the addSiteModal handling an onClose event 1`] = `
<div>
  <div
    className="hBejNq my-yoast-modal__content"
    contentLabel="Add a new site"
    id="Modal"
    isOpen={true}
    onRequestClose={[Function]}
    overlayClassName="hBejNq my-yoast-modal__overlay"
    role="dialog">
    <div
      className="hfMDPK">
      <h1
        className="dQgKWp">
        <span>
          Add Site
        </span>
      </h1>
      <p
        className="hUuGBg">
        <label
          htmlFor="addSiteInputField">
          <span>
            Please enter the URL of the site you would like to link with your account:
          </span>
        </label>
      </p>
      <input
        aria-describedby="addSiteInfo"
        className="jssTrf"
        defaultValue=""
        id="addSiteInputField"
        onChange={[Function]}
        placeholder="https://example-site.com"
        type="url" />
      <div
        className="gCxKqx">
        <div
          className="iMrCDo">
          <button
            className="cAeNWO"
            onClick={[Function]}
            type="button">
            <span>
              cancel
            </span>
          </button>
          <button
            className="gxQXtW"
            onClick={[Function]}
            type="button">
            <span>
              connect
            </span>
          </button>
        </div>
      </div>
      <div
        className="bPgglV" />
      <img
        alt=""
        className="ireVCO"
        src="addsite.svg" />
    </div>
  </div>
</div>
`;

exports[`test the addSiteModal handling an onClose event 2`] = `
<div>
  <div
    className="hBejNq my-yoast-modal__content"
    contentLabel="Add a new site"
    id="Modal"
    isOpen={true}
    onRequestClose={[Function]}
    overlayClassName="hBejNq my-yoast-modal__overlay"
    role="dialog">
    <div
      className="hfMDPK">
      <h1
        className="dQgKWp">
        <span>
          Add Site
        </span>
      </h1>
      <p
        className="hUuGBg">
        <label
          htmlFor="addSiteInputField">
          <span>
            Please enter the URL of the site you would like to link with your account:
          </span>
        </label>
      </p>
      <input
        aria-describedby="addSiteInfo"
        className="jssTrf"
        defaultValue=""
        id="addSiteInputField"
        onChange={[Function]}
        placeholder="https://example-site.com"
        type="url" />
      <div
        className="gCxKqx">
        <div
          className="iMrCDo">
          <button
            className="cAeNWO"
            onClick={[Function]}
            type="button">
            <span>
              cancel
            </span>
          </button>
          <button
            className="gxQXtW"
            onClick={[Function]}
            type="button">
            <span>
              connect
            </span>
          </button>
        </div>
      </div>
      <div
        className="bPgglV" />
      <img
        alt=""
        className="ireVCO"
        src="addsite.svg" />
    </div>
  </div>
</div>
`;

exports[`test the addSiteModal handling an onConnect event 1`] = `
<div>
  <div
    className="hBejNq my-yoast-modal__content"
    contentLabel="Add a new site"
    id="Modal"
    isOpen={true}
    onRequestClose={[Function]}
    overlayClassName="hBejNq my-yoast-modal__overlay"
    role="dialog">
    <div
      className="hfMDPK">
      <h1
        className="dQgKWp">
        <span>
          Add Site
        </span>
      </h1>
      <p
        className="hUuGBg">
        <label
          htmlFor="addSiteInputField">
          <span>
            Please enter the URL of the site you would like to link with your account:
          </span>
        </label>
      </p>
      <input
        aria-describedby="addSiteInfo"
        className="jssTrf"
        defaultValue=""
        id="addSiteInputField"
        onChange={[Function]}
        placeholder="https://example-site.com"
        type="url" />
      <div
        className="gCxKqx">
        <div
          className="iMrCDo">
          <button
            className="cAeNWO"
            onClick={[Function]}
            type="button">
            <span>
              cancel
            </span>
          </button>
          <button
            className="gxQXtW"
            onClick={[Function]}
            type="button">
            <span>
              connect
            </span>
          </button>
        </div>
      </div>
      <div
        className="bPgglV" />
      <img
        alt=""
        className="ireVCO"
        src="addsite.svg" />
    </div>
  </div>
</div>
`;

exports[`test the addSiteModal handling an onConnect event 2`] = `
<div>
  <div
    className="hBejNq my-yoast-modal__content"
    contentLabel="Add a new site"
    id="Modal"
    isOpen={true}
    onRequestClose={[Function]}
    overlayClassName="hBejNq my-yoast-modal__overlay"
    role="dialog">
    <div
      className="hfMDPK">
      <h1
        className="dQgKWp">
        <span>
          Add Site
        </span>
      </h1>
      <p
        className="hUuGBg">
        <label
          htmlFor="addSiteInputField">
          <span>
            Please enter the URL of the site you would like to link with your account:
          </span>
        </label>
      </p>
      <input
        aria-describedby="addSiteInfo"
        className="jssTrf"
        defaultValue=""
        id="addSiteInputField"
        onChange={[Function]}
        placeholder="https://example-site.com"
        type="url" />
      <div
        className="gCxKqx">
        <div
          className="iMrCDo">
          <button
            className="cAeNWO"
            onClick={[Function]}
            type="button">
            <span>
              cancel
            </span>
          </button>
          <button
            className="gxQXtW"
            onClick={[Function]}
            type="button">
            <span>
              connect
            </span>
          </button>
        </div>
      </div>
      <div
        className="bPgglV" />
      <img
        alt=""
        className="ireVCO"
        src="addsite.svg" />
    </div>
  </div>
</div>
`;
